crates/runtime/src/http/v1.rs

```diff
@@ -53,64 +53,208 @@ pub(crate) mod inference {
+            Ok(inference_result) => {
+                if let Some(column_data) = inference_result.column_by_name("y") {
+                    if let Some(array) = column_data.as_any().downcast_ref::<Float32Array>() {
                         let result = array.values().iter().copied().collect_vec();
-                        (
-                            StatusCode::OK,
-                            Json(ForecastResponse {
-                                forecast: result,
-                                duration_ms: start_time.elapsed().as_millis(),
-                            }),
-                        )
-                            .into_response()
+                        return PredictResponse {
+                            status: PredictStatus::Success,
+                            error_message: None,
+                            model_name,
+                            lookback,
+                            forecast: result,
+                            duration_ms: start_time.elapsed().as_millis(),
+                        };
                     }
-                    None => (StatusCode::INTERNAL_SERVER_ERROR,).into_response(),
-                },
-                None => (StatusCode::INTERNAL_SERVER_ERROR,).into_response(),
-            },
+                    tracing::error!(
+                        "Unable to cast inference result for model {model_name} to Float32Array: {column_data:?}"
+                    );
+                    return PredictResponse {
+                        status: PredictStatus::InternalError,
+                        error_message: Some(
+                            "Unable to cast inference result to Float32Array".to_string(),
+                        ),
+                        model_name,
+                        lookback,
+                        forecast: vec![],
+                        duration_ms: start_time.elapsed().as_millis(),
+                    };
+                }
+                tracing::error!(
+                    "Unable to find column 'y' in inference result for model {model_name}"
+                );
+                PredictResponse {
+                    status: PredictStatus::InternalError,
+                    error_message: Some(
+                        "Unable to find column 'y' in inference result".to_string(),
+                    ),
+                    model_name,
+                    lookback,
+                    forecast: vec![],
+                    duration_ms: start_time.elapsed().as_millis(),
+                }
+            }
             Err(e) => {
-                tracing::error!("Unable to run inference: {}", e);
-                (StatusCode::INTERNAL_SERVER_ERROR,).into_response()
+                tracing::error!("Unable to run inference: {e}");
+                PredictResponse {
+                    status: PredictStatus::InternalError,
+                    error_message: Some(e.to_string()),
+                    model_name,
```

Comment:
@mitchdevenport can we get model version in here?

Reply:
Yes, though it might be confusing because locally the user can name the model whatever they want.

Should we add the full `from` reference to disambiguate?

`spice.ai/lukekim/demo/models/darts:latest`

Reply:
Actually, I think it makes sense because the user knows what they named it locally, we can always easily add the full reference later